### ISSUE TITLE
WP-CLI image creating or updating images

### ISSUE THREAD

**Issue by ShawnJW** [woocommerce/woocommerce#21419]
The bug reported here https://github.com/woocommerce/woocommerce/issues/18964  still seems to be unresolved even though it's marked closed, I must be missing something.  I am running the latest WordPress and WooCommerce builds.

**To Reproduce**
Using either 
1. wp wc product create --images='[ { "id" : 1 } ]'
2. wp post meta update 1 _thumbnail_id 1
3 wp wc product update 18334  --images='[ { "id" : 788643.jpg } ]'

Produces no featured image for the product post, even when the CLI returns a "Success" message.

**Expected behavior**
I expect to pull the image Id from the media library based on the id, in the example 1 = 1.jpg

**Isolating the problem (mark completed items with an [x]):**
- [ x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [ x] I can reproduce this bug consistently using the steps above.

**WordPress Environment**
<details>
```
Copy and paste the system status report from **WooCommerce > System Status** in WordPress admin.
```

WooCommerce version: |   | 3.4.5
Log directory writable: |   | /Applications/MAMP/htdocs/sourcemolecular/wp-content/uploads/wc-logs/
WordPress version: |   | 4.9.8
WordPress multisite: |   | –
WordPress memory limit: |   | 15 GB
WordPress debug mode: |   |  
WordPress cron: |   |  
Language: |   | en_US
External object cache: |   | –

### WordPress Environment ###

WC Version: 3.4.5
Log Directory Writable: ✔
WP Version: 4.9.8
WP Multisite: –
WP Memory Limit: 15 GB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache
PHP Version: 7.1.12 - We recommend using PHP version 7.2 or above for greater performance and security. How to update your PHP version
PHP Post Max Size: 32 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.52.1
OpenSSL/1.0.2j

SUHOSIN Installed: –
MySQL Version: 5.6.38
Max Upload Size: 32 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.4.5
WC Database Prefix: wp_
MaxMind GeoIP Database: ✔
Total Database Size: 114.14MB
Database Data Size: 78.88MB
Database Index Size: 35.26MB

### Security ###

Secure connection (HTTPS): ❌
					Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ❌Error messages should not be shown to visitors.

### Active Plugins (1) ###

WooCommerce: by Automattic – 3.4.5

### Theme ###

Name: Twenty Seventeen
Version: 1.7
Author URL: https://wordpress.org/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –
`
</details>

**Comment by claudiulodro** [comment]
Hi there.

It's closed because the problem is a WordPress core issue, and it is going to be fixed in WordPress core in the 5.0 release which is not out yet. In the meantime, I recommend using the workaround provided in that issue. Hope this clears things up.

Thanks.

**Comment by Deichscheich** [comment]
> Hi there.
> 
> It's closed because the problem is a WordPress core issue, and it is going to be fixed in WordPress core in the 5.0 release which is not out yet. In the meantime, I recommend using the workaround provided in that issue. Hope this clears things up.
> 
> Thanks.

@claudiulodro, @ShawnJW, has this in fact been fixed? Using 5.3.2, this command results only in a placeholder image (id 0)
wp wc product create --images='[ { "id" : 34 } ]'

You "recommend using the workaround provided in that issue", and I understand the workaround to be to define an object like above. Please correct me if I misunderstood. What is the workaround?

#18964

**Comment by Deichscheich** [comment]
:(
Sorry, my bad. It does work.
